src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java

```diff
@@ -530,9 +536,14 @@ private static String quoteParams(String... params) {
         }
     }
 
+    /** Explicitly pass in name used to generate variable ID because methodName is not always valid */
+    private Tuple<String, String> dateFunctionTemplate(String name, String methodName, SQLExpr field) {
+        String id = nextId(name);
+        return new Tuple<>(id, def(id, doc(field) + ".value." + methodName));
+    }
+
```

Comment:
Sorry, not very clear about this change. Can you give an example how this affects things.

Reply:
Sorry for confusion. Basically this is caused by the painless API changes above, for example:

```
functionStr = dateFunctionTemplate("weekOfWeekyear",
                                   "get(WeekFields.ISO.weekOfWeekBasedYear())",
                                   (SQLExpr) paramers.get(0).value);
```

The old `dateFunctionTemplate(methodName, field)` passes `get(WeekFields.ISO.weekOfWeekBasedYear())` to `nextId()` which generates an invalid variable name in painless script. Will add a little more elaboration in JavaDoc.

Reply:
Thanks, clear now.

Reply:
Cool. Thanks!